msm/skill_entry.py

```diff
@@ -208,6 +211,19 @@ def update(self):
 
         try:
             git.fetch()
+            current_branch = git.rev_parse('--abbrev-ref', 'HEAD').strip()
+            if current_branch in UNSAFE_BRANCHES:
+                # Make sure there's no modifications to tracked files
+                if git.status(porcelain=True, untracked='no') != '':
+                    raise SkillModified('Uncommitted changes, aborting')
+                # Check out correct branch
+                sha_branch = git.branch(contains=self.sha).split('\n')[0]
+                sha_branch = sha_branch[1:].strip()
+                git.checkout(sha_branch)
+                # Reset to common ancestor to make the merge work OK
+                common_sha = git.merge_base(sha_before, self.sha)
+                git.reset(common_sha, hard=True)
```

Comment:
Any reason we need to hard reset once we've switched branches? This would delete any new commits on `18.02`. ie.:
```
git checkout 18.02
git commit -a -m "Add feature"
git checkout master
msm update
# Now the "Add feature" commit is lost
```